### PR TITLE
SW-4329 Assign short numbers to facilities

### PR DESCRIPTION
**Pull request by sgrimm**
To support generating nursery batch numbers that include an indication of which
nursery the batch is in, add the concept of a "facility number." Facility numbers
are per-organization, per-facility-type. That is, for each organization, its first
nursery will have a facility number of 1, and its first seed bank will also have
facility number 1.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1410** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1410" target="_blank">Graphite</a>  👈
    * **PR #1411** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1411" target="_blank">Graphite</a> 
      * **PR #1415** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1415" target="_blank">Graphite</a> 
        * **PR #1416** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1416" target="_blank">Graphite</a> 
      * **PR #1412** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1412" target="_blank">Graphite</a> 
        * **PR #1413** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1413" target="_blank">Graphite</a> 
          * **PR #1414** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1414" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1410?utm_source=stack-comment).

File: src/main/resources/db/migration/0200/V213__FacilityNumber.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+ALTER TABLE facilities ADD COLUMN facility_number INTEGER;
+
+WITH facility_numbers AS (
+    SELECT id, RANK() OVER (PARTITION BY organization_id, type_id ORDER BY id) AS facility_number
```
Comment (karthikbtf) [inline src/main/resources/db/migration/0200/V213__FacilityNumber.sql:4]: Nice, my understanding is the rank + partition criteria produces ranks that are the correct facility numbers.

**Review by karthikbtf — APPROVED** [review]
Looks great 👍

File: src/main/resources/db/migration/0200/V213__FacilityNumber.sql (hunk 1 above)
Comment (sgrimm) [reply src/main/resources/db/migration/0200/V213__FacilityNumber.sql:4]: Yeah, `RANK()` returns a 1-indexed sequence number within each partition. A bit more explanation if you're curious: https://www.sqltutorial.org/sql-window-functions/sql-rank/

**Comment by sgrimm** [comment]
### Merge activity

* **Oct 23, 12:35 PM**: @sgrimm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1410).
* **Oct 23, 12:36 PM**: @sgrimm merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1410).

**Pull request merged**